oci-registry/build-multi-arch.sh

```diff
@@ -0,0 +1,80 @@
+#!/bin/sh
+
+#
+# Copyright Red Hat
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+buildfolder="$(basename "$(dirname "$0")")"
+# Due to command differences between podman and docker we need to separate the process
+# for creating and adding images to a multi-arch manifest
+podman=${USE_PODMAN:-false}
+# Stores all created image tags 
+images=()
+# Base Repository
+BASE_REPO="quay.io/devfile/oci-registry"
+BASE_TAG="next"
+
+function build {
+    IMAGE="$BASE_REPO:$2"
+
+    echo "Building: ${IMAGE}"
+    $1 build -t $IMAGE --platform "linux/$2" "$buildfolder"
+
+    echo "Tagging: ${IMAGE}"
+    $1 tag "$IMAGE" "$IMAGE"
+
+    echo "Pushing: ${IMAGE}"
+    $1 push "$IMAGE"
+
+    # Add image to list of all images to be added to a manifest
+    images+=("${IMAGE}")
+}
+
+function engine-handler {
+    for arch in amd64 arm64 ; do
+        build "$1" "$arch"
+    done
+}
+
+
+if [ ${podman} == true ]; then
+  echo "Executing with podman"
+
+  # Build and push multi-arch images
+  engine-handler podman
+
+  # Create manifest and add images
+  podman manifest create oci-registry-manifest
+  for img in "${images[@]}" ; do
+    podman manifest add oci-registry-manifest "$img"
+  done
+
+  # Push and delete local manifest
+  podman manifest push oci-registry-manifest "$BASE_REPO":"$BASE_TAG"
+  podman manifest rm oci-registry-manifest
+
+else
+  echo "Executing with docker"
+
+  # Build and push multi-arch images
+  engine-handler docker
```

Comment:
I was testing locally the script (replaced with my quay.io workspace) and I'm getting
```
Dockerfile:18
--------------------
  16 |     
  17 |     FROM registry.access.redhat.com/ubi8-minimal:8.2
  18 | >>> RUN microdnf update -y && rm -rf /var/cache/yum && microdnf install ca-certificates httpd-tools
  19 |     
  20 |     # Create a non-root user to run the server as
--------------------
ERROR: failed to solve: process "/bin/sh -c microdnf update -y && rm -rf /var/cache/yum && microdnf install ca-certificates httpd-tools" did not complete successfully: exit code: 1
```

Reply:
Not sure tho if is something I'm missing. Tried:

* bash oci-registry/build-multi-arch.sh amd64
* bash oci-registry/build-multi-arch.sh

Reply:
This might be a local issue with your Docker, the same thing was happening until I added QEMU for Docker here: https://github.com/devfile/registry-support/pull/240/commits/d8e62391a01b68ba740c54faa831ad87a260cb64

I believe it is an issue with emulation or something along those lines, can you run `podman run --rm --privileged tonistiigi/binfmt` or `docker run --rm --privileged tonistiigi/binfmt` and check what architectures are under the "supported" section like so:
```
"supported": [
    "linux/arm64",
    "linux/amd64",
    "linux/riscv64",
    "linux/ppc64le",
    "linux/s390x",
    "linux/386",
    "linux/mips64le",
    "linux/mips64"
  ]
```

Reply:
Yeap adding QEMU works :) Do you think it makes sense to add this information somewhere as troubleshooting in case more people face the same issue?

Reply:
Awesome I'm glad that fixed the issue! Since the `build-multi-arch.sh` was initially designed for being run as part of our CI I didn't think to add a readme or anything related to it, but it may be a good idea incase someone wants to run locally. Do you think including a section in the `oci-registry/README` or a comment within `oci-registry/build-multi-arch.sh` itself would be best for this situation?

Reply:
Yeah maybe a testing locally section or something like this I think is sufficient.

Reply:
In order to check the support I've followed the command you've shared. I've installed QEMU with this command:

```
sudo apt-get install -y gcc-arm-linux-gnueabihf libc6-dev-armhf-cross qemu-user-static qemu-system-i386
```